src/screens/smart-screen/smart-screen.tsx

```diff
@@ -0,0 +1,216 @@
+            pubKey: this.props.walletPublicKey,
+            blockchain: this.props.account.blockchain,
+            chainId: this.props.chainId,
+            address: this.props.account.address,
+            tab: this.props.context?.tab
+        });
+
+        if (dashboard && dashboard[screenKey]) {
+            const screenData: IScreenData = dashboard[screenKey];
+
+            screenData &&
+                this.setState({
+                    screenData,
+                    isLoading: screenData.isLoading,
+                    error: screenData.error
+                });
+        } else if (token && token[screenKey]) {
+            const screenData: IScreenData = token[screenKey];
+
+            screenData &&
+                this.setState({
+                    screenData,
+                    isLoading: screenData.isLoading,
+                    error: screenData.error
+                });
+        } else {
+            this.setState({
+                screenData: undefined,
+                isLoading: true,
+                error: undefined
+            });
+        }
+    }
+
+    public render() {
+        const { styles, theme } = this.props;
+        const { screenData } = this.state;
+
+        if (this.state.isLoading) {
+            return (
+                <View style={styles.skeletonWrapper}>
+                    {new Array(4).fill('').map((_, index: number) => (
+                        <SkeletonPlaceholder
+                            key={`skelet-${index}`}
+                            backgroundColor={theme.colors.textTertiary}
+                            highlightColor={theme.colors.accent}
+                            speed={Math.floor(Math.random() * 700) + 1000}
+                        >
+                            <View style={styles.detailsSkeletonComp}>
+                                <View style={styles.detailsSkeletonIcon} />
+                                <View style={{ justifyContent: 'space-between' }}>
+                                    <View style={styles.detailsSkeletonPrimaryValue} />
+                                    <View style={styles.detailsSkeletonSecondaryValue} />
+                                </View>
+                            </View>
+                        </SkeletonPlaceholder>
+                    ))}
+                </View>
+            );
+        }
+
+        if (screenData?.response?.widgets) {
+            return (
+                <View>
+                    <Widgets
+                        data={screenData.response.widgets}
+                        actions={{
+                            claimRewardNoInput: this.props.claimRewardNoInput,
+                            withdraw: this.props.withdraw
+                        }}
+                        account={this.props.account}
+                        chainId={this.props.chainId}
+                    />
+                </View>
+            );
+        }
+
+        if (this.state.error) {
```

Comment:
We have to decide if we should display error if we have something in screen data, yes the data could be old but maybe it's better to display data instead of an error... ask @i3uRi about this...